src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -168,6 +169,10 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
           i18n: { key: 'oform.errorbanner.title' },
         } as IdxMessage);
       }
+      // clear the userIdentifier value when returning to the identify flow
+      if (newTransaction.nextStep?.name === IDX_STEP.IDENTIFY) {
+        setUserIdentifier(null);
+      }
```

Comment:
It might be better to put this setter somewhere where we react to the transaction changing, rather than where we are setting the changed transaction, if that makes sense. I view this hook as a place where we are updating the transaction value, whereas the bit about clearing the userIdentifier value is a reaction to when that changes.

We have some effect hooks that are in `Widget/index` that might be more appropriate since they are run whenever transaction changes.

Reply:
Okay yeah I see what youre saying.  The `useOnSubmit` seemed like a reliable place to put it but the `Widget/index` is probably more appropriate.